cc/private/rules_impl/cc_library.bzl

```diff
@@ -238,14 +238,25 @@ def _cc_library_impl(ctx):
         libraries = libraries_to_link,
     )
 
+    supports_start_end_lib = cc_common.is_enabled(
+        feature_configuration = feature_configuration,
+        feature_name = "supports_start_end_lib",
+    )
+
     files_builder = []
     if linking_outputs.library_to_link != None:
         artifacts_to_build = linking_outputs.library_to_link
-        if artifacts_to_build.static_library != None:
-            files_builder.append(artifacts_to_build.static_library)
+        if supports_start_end_lib and (artifacts_to_build.objects or artifacts_to_build.pic_objects):
+            if artifacts_to_build.pic_objects:
+                files_builder.extend(artifacts_to_build.pic_objects)
+            elif artifacts_to_build.objects:
```

Comment:
Previously both the pic and non-pic version were built, now only one of the two flavors is requested. There are pros and cons to both approaches, I guess. We should at least leave a comment explaining why this differs between the two feature states.

Reply:
I was wondering why that was the case, or in what configurations you would end up with both, but I can play it safe and mirror today's behavior 🤔 , not sure what the ideal is?